src/AzureRMTools.ts

```diff
@@ -209,20 +213,68 @@ export class AzureRMTools {
                 diagnostics.push(this.getVSCodeDiagnosticFromIssue(deploymentTemplate, warning, vscode.DiagnosticSeverity.Warning));
             }
 
-            if (ext.addCompletionDiagnostic) {
-                // Add a diagnostic to indicate expression validation is done (for testing)
-                diagnostics.push(<vscode.Diagnostic>{
-                    severity: vscode.DiagnosticSeverity.Information,
-                    message: expressionsDiagnosticsCompletionMessage,
-                    source: expressionsDiagnosticsSource,
-                    code: ""
-                });
+            let languageServerLanguageDiagnostic = this.getLanguageServerStateDiagnostic();
+            if (languageServerLanguageDiagnostic) {
+                diagnostics.push(languageServerLanguageDiagnostic);
+            }
+
+            let completionDiagnostic = this.getCompletionDiagnostic();
+            if (completionDiagnostic) {
+                diagnostics.push(completionDiagnostic);
             }
 
             this._diagnosticsCollection.set(document.uri, diagnostics);
```

Comment:
Curious: Do you guys maintain your own diagnostics collection? The `LanguageServerClient` can handle this transparently and provide access to the underlying diagnostics collection

Reply:
Yes, we do.  I did notice that recently, might look into merging them, although there appears to be no harm in having two separate ones. Thx.